lib/ansible/modules/windows/win_firewall.ps1

```diff
@@ -0,0 +1,79 @@
+#!powershell
+# This file is part of Ansible
+
+# Copyright 2017, Michael Eaton <meaton@iforium.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+
+$result = @{
+    changed = $false
+}
+
+# get params
+$params = Parse-Args $args -supports_check_mode $false
+
+$profile = Get-AnsibleParam -obj $params -name "profile" -type "str" -failifempty $true -validateset 'Public','Domain','Private'
+$wantedstate = Get-AnsibleParam -obj $params -name "state" -type "str" -failifempty $true -validateset 'enabled', 'disabled'
+
+Try {
+
+  # get current profile status
+
+  $currentstate = (Get-NetFirewallProfile -Name $profile).Enabled
+
+  # to be enabled
+  if ($wantedstate -eq 'enabled')
+  {
+    if ($currentstate -eq $true)
+    {
+        $result.changed = $false;
+        $result.profile = $profile;
+        $result.enabled = [System.Convert]::ToBoolean($currentstate);
+        Exit-Json $result
```

Comment:
This can be removed (final Exit-Json picks this up).

In the end, the whole block can be replaced with a single:
```powershell
# Nothing to do.
```